src/upgrade.js

```diff
@@ -62,8 +62,19 @@ Upgrade.getAll = async function () {
 Upgrade.appendPluginScripts = async function (files) {
 	// Find all active plugins
 	const activePlugins = await plugins.getActive();
-	activePlugins.forEach((plugin) => {
-		const configPath = path.join(paths.nodeModules, plugin, 'plugin.json');
+	for (const plugin of activePlugins) {
+		let configPath = path.join(paths.nodeModules, plugin, 'plugin.json');
+		// eslint-disable-next-line no-await-in-loop
+		if (!await file.exists(configPath)) {
+			configPath = path.join(paths.nodeModules, plugin, 'plugin.js');
+		}
+		// eslint-disable-next-line no-await-in-loop
+		if (!await file.exists(configPath)) {
```

Comment:
This if block should move into the above if block, otherwise if there is a plugin.json we are doing 2x file.exists(configPath) checks with the same `configPath` value.